language/stdlib/transaction_scripts/add_currency_to_account.move

```diff
@@ -1,9 +1,29 @@
 script {
 use 0x1::LibraAccount;
+// Prover deps:
+use 0x1::Libra;
+use 0x1::Signer;
 
-/// Add the currency identified by the type `currency` to the sending accounts.
-/// Aborts if the account already holds a balance fo `currency` type.
+/// Add a `Currency` balance to `account`, which will enable `account` to send and receive
+/// `Libra<Currency>`.
+/// Aborts with NOT_A_CURRENCY if `Currency` is not an accepted currency type in the Libra system
+/// Aborts with `LibraAccount::ADD_EXISTING_CURRENCY` if the account already holds a balance in
+/// `Currency`.
+/// Aborts with `LibraAccount::PARENT_VASP_CURRENCY_LIMITS_DNE` if `account` is a `ChildVASP` whose
+/// parent does not have an `AccountLimits<Currency>` resource.
 fun add_currency_to_account<Currency>(account: &signer) {
     LibraAccount::add_currency<Currency>(account);
 }
+spec fun add_currency_to_account {
+    pragma verify = true;
+    pragma aborts_if_is_partial = true;
```

Comment:
I understand why you'd use this, but just for clarification, goal should be eventually to avoid this. We want that all aborts are covered, unless they are programmer errors. Otherwise the aborts_if specs have diminished value. To distinguish programmer from user errors, we would like to see a partitioning of the error space at some point in the future. Also would like to have the convention that any implicit aborts should be considered a programmer error. E.g.:

```
assert(!exists<R>(a), Errrors::RESOURCE_EXISTS);
move_to(a, R{...}); /// This would produce Errors::UNEXPECTED_RESOURCE_EXISTS or similar
```

Reply:
Agreed with both conventions! It hurt me to use `aborts_if_partial` because I was very interested to see what other error cases I might be missing...